test/studies/lcals/LCALSMain.chpl

```diff
@@ -0,0 +1,1203 @@
+    var ref_loop_stat = new LoopStat(LoopLength.NUM_LENGTHS);
+    suite_info.ref_loop_stat = ref_loop_stat;
+
+    ref_loop_stat.loop_length[LoopLength.LONG]        = 24336;
+    ref_loop_stat.loop_length[LoopLength.MEDIUM]      = 3844;
+    ref_loop_stat.loop_length[LoopLength.SHORT]       = 64;
+    ref_loop_stat.samples_per_pass[LoopLength.LONG]   = 30000;
+    ref_loop_stat.samples_per_pass[LoopLength.MEDIUM] = 300000;
+    ref_loop_stat.samples_per_pass[LoopLength.SHORT]  = 50000000;
+  }
+
+  proc defineLoopSuiteRunInfo(run_variants, run_loop,
+                              sample_frac:real, loop_length_factor:real) {
+
+    var suite_info = getLoopSuiteRunInfo();
+    var run_variant_names = getVariantNames(run_variants);
+
+    writeln("\n defineLoopSuiteRunInfo...");
+
+    suite_info.loop_samp_frac = sample_frac;
+
+    suite_info.loop_weights.resize(WeightGroup.NUM_WEIGHT_GROUPS);
+    suite_info.loop_weights[WeightGroup.DATA_PARALLEL] = 2.0;
+    suite_info.loop_weights[WeightGroup.ORDER_DEPENDENT] = 1.8;
+    suite_info.loop_weights[WeightGroup.TRANSCENDENTAL] = 1.7;
+    suite_info.loop_weights[WeightGroup.DATA_DEPENDENT] = 1.7;
+    suite_info.loop_weights[WeightGroup.POINTER_NEST] = 1.4;
+    suite_info.loop_weights[WeightGroup.COMPLEX] = 1.0;
+
+    suite_info.loop_length_names.resize(LoopLength.NUM_LENGTHS);
+    suite_info.loop_length_names[LoopLength.LONG] = "LONG";
+    suite_info.loop_length_names[LoopLength.MEDIUM] = "MEDIUM";
+    suite_info.loop_length_names[LoopLength.SHORT] = "SHORT";
+
+    suite_info.num_loops_run.resize(run_variant_names.size());
+    suite_info.tot_time.resize(run_variant_names.size());
+    suite_info.fom_rel.resize(run_variant_names.size());
+    suite_info.fom_rate.resize(run_variant_names.size());
+
+    for ilv in 0..#run_variant_names.size() {
+      suite_info.addLoopStats(run_variant_names[ilv]);
+
+      if suite_info.num_loops_run[ilv] == nil then
+        suite_info.num_loops_run[ilv] = new vector(int);
+      suite_info.num_loops_run[ilv].resize(LoopLength.NUM_LENGTHS);
+
+      if suite_info.tot_time[ilv] == nil then
+        suite_info.tot_time[ilv] = new vector(real);
+      suite_info.tot_time[ilv].resize(LoopLength.NUM_LENGTHS);
+
+      if suite_info.fom_rel[ilv] == nil then
+        suite_info.fom_rel[ilv] = new vector(real);
+      suite_info.fom_rel[ilv].resize(LoopLength.NUM_LENGTHS);
+
+      if suite_info.fom_rate[ilv] == nil then
+        suite_info.fom_rate[ilv] = new vector(real);
+      suite_info.fom_rate[ilv].resize(LoopLength.NUM_LENGTHS);
+    }
+
+    var shared_loop_length = new vector(int);
+    shared_loop_length.resize(LoopLength.NUM_LENGTHS);
+    shared_loop_length[LoopLength.LONG]   = (44217 * loop_length_factor):int;
+    shared_loop_length[LoopLength.MEDIUM] = (5001 * loop_length_factor):int;
+    shared_loop_length[LoopLength.SHORT]  = (171 * loop_length_factor):int;
+
+    // ADomain::loop_length_factor = loop_length_factor;
```

Comment:
This is another place where I need to look at loop_length_factor variables more closely.  The field that was static in ADomain in the reference needs to have all instances set to the module-level const here.  Probably the easiest way to do it will be to add an argument to the ADomain constructor and pass the module-level const to it rather than changing the static member here.